pkg/filestore/writer.go

```diff
@@ -12,22 +12,25 @@ import (
 	"github.com/cert-manager/csi-lib/storage"
 
 	"github.com/jetstack/cert-manager-csi/pkg/apis/defaults"
-	"github.com/jetstack/cert-manager-csi/pkg/apis/v1alpha1"
+	csiapi "github.com/jetstack/cert-manager-csi/pkg/apis/v1alpha1"
 	"github.com/jetstack/cert-manager-csi/pkg/apis/validation"
 )
 
-// Writer wraps the storage backend to allow access for writing data
+// Writer wraps the storage backend to allow access for writing data.
 type Writer struct {
 	Store storage.Interface
 }
 
+// WriteKeypair writes the given certificate, CA, and private key data to their
+// respective file locations, according to the volume attributes. Also writes
+// or updates the metadata file, including a calculated NextIssuanceTime.
 func (w *Writer) WriteKeypair(meta metadata.Metadata, key crypto.PrivateKey, chain []byte, ca []byte) error {
```

Comment:
Question: you accept crypto.PrivateKey into this function but assert that it is an *rsa.PrivateKey further in. Does this cause a crash if you pass in other types of private key?

Reply:
We currently only support RSA keys. We should probably expose some options for generating other types of keys.

Reply:
@jakexks there will only ever be an rsa.PrivateKey passed in as it is the only type supported - this method _has to_ accept a `crypto.PrivateKey` to satisfy the corresponding interface in `csi-lib` :)